src/server/event.recorder.ts

```diff
@@ -0,0 +1,343 @@
+import {
+    EventSelector,
+    PrebootOptions,
+    PrebootAppData,
+    PrebootData,
+    Element,
+    DomEvent,
+    Window,
+    Document,
+    Selection,
+    ServerClientRoot,
+    getNodeKeyForPreboot
+} from '../common';
+
+/**
+ * Called right away to initialize preboot
+ *
+ * @param opts All the preboot options
+ */
+export function init(opts: PrebootOptions, win?: Window) {
```

Comment:
Note that these functions are structured like this for testing purposes, but when we generate the inline version of preboot each of these functions are pulled in and `toString()` is called on them in order to generate the inline code. SO, important that nothing imported other than interfaces here AND no variables referenced in this module (i.e. only isolated functions that don't have shared state).